core/src/storage2/collections/binary_heap/mod.rs

```diff
@@ -337,11 +337,12 @@ where
         let cur = self.begin + n;
         self.begin += 1 + n;
 
-        let group_index = group::get_group_index(cur);
-        self.elems
-            .get(group_index)
-            .expect("access is out of bounds")
-            .as_ref(group_index)
+        let children_index = children::get_children_storage_index(cur);
+        let children = self
+            .elems
+            .get(children_index)
+            .expect("access is out of bounds");
+        children.child(children::get_child_pos(children_index))
```

Comment:
This API to `children` still seems clunky to me. Maybe a single method on `Elements` something like (pseudocode) which might avoid having to do all these `children` calls everywhere

```
fn get_child(&self, index: u32) -> Option<(T, ChildPosition), Children<T>)> { .. }